Sources/NIOEchoClient/main.swift

```diff
@@ -21,13 +21,22 @@ private final class EchoHandler: ChannelInboundHandler {
     public typealias OutboundOut = ByteBuffer
     private var numBytes = 0
 
+
     public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
-        numBytes -= self.unwrapInboundIn(data).readableBytes
+
+        var byteBuffer = self.unwrapInboundIn(data) as ByteBuffer
+        numBytes -= byteBuffer.readableBytes
 
         assert(numBytes >= 0)
 
         if numBytes == 0 {
-            print("Received the line back from the server, closing channel")
+            if let charArray = byteBuffer.readBytes(length: byteBuffer.readableBytes) {
+                var str = ""
+                for byte in charArray {
+                    str += String(UnicodeScalar(byte))
+                }
```

Comment:
Ah, with this tighter loop it's a lot easier to see what we're getting into here. This can all be replaced with `if let string = byteBuffer.readString(length: byteBuffer.readableBytes)`. We'll also want an error message for the non-bound branch.

Reply:
Thank you!  Using `if let string = byteBuffer.readString(length: byteBuffer.readableBytes)` is a lot more concise than using the loop I have above.